### ISSUE TITLE
False solid-reactivity error for a non-sourced, initialized createResource

### ISSUE THREAD

**Issue by jb-1980** [solidjs-community/eslint-plugin-solid#199]
**Describe the bug**
When using create resource like `const [data] = createResource(asyncFetcher, options)`, there is a lint error: "This tracked scope should not be async."

**To Reproduce**
Add code similar to the description, and note the lint error:
```ts
const [intializedResource] = createResource(async () => [], {
    initialValue: [],
});
```
See https://playground.solidjs.com/anonymous/b67979b9-c72b-4b52-91e1-dcec2b445ade for a more extensive example.

**Expected behavior**
When the fetcher function does not contain reactive values, there should be no lint error. `createResource(nonReactiveFetcher, options)` is a legit use of `createResource`.

**Environment (please complete the following information):**

- OS: [Windows 11, Ubuntu 24.04]
- Node version (`node --version`): v22
- `eslint-plugin-solid` version (`npm list eslint-plugin-solid`/`yarn why eslint-plugin-solid`): 0.14.5 and whatever is in solid playground
- `eslint` version (`npm list eslint`/`yarn why eslint`): 9.20.1 and whatever is in solid playground

**Additional context**
I believe I can see the source of the issue.[ L970 of rules/reactivity.ts](https://github.com/solidjs-community/eslint-plugin-solid/blob/8a17adecb2451bb1cacf8add07d68f2a49bb3614/packages/eslint-plugin-solid/src/rules/reactivity.ts#L970) has this in the conditional:
```ts
(matchImport("createResource", callee.name) && node.arguments.length >= 2)
```
Since the `arguments.length` of `createResource(fetcher, opts)` is 2, this state is incorrectly pushed into the tracked scopes.

I believe the solution will require a check similar to [Solid's implementation of `createResource`](https://github.com/solidjs/solid/blob/main/packages/solid/src/reactive/signal.ts#L607):
`if (typeof node.arguments[1] === "function") { /* There is a source */ }`

So the solution may be as simple as
```ts
(matchImport("createResource", callee.name) && typeof node.arguments[1] === "function")
```

- [x] I would be willing to contribute a PR to fix this issue

**Comment by ryansolid** [comment]
Confirmed on current main: `createResource(asyncFetcher, options)` (no source) still reports `noAsyncTrackedScope` — the rule misreads the options object as a fetcher and the fetcher as a source. Queued as a 0.16.1 false-positive fix, together with #195. (Solid 2.0 note: `createResource` no longer exists there, so this fix applies to 1.x-mode linting.)

**Comment by ryansolid** [comment]
Fixed in v0.16.1. `createResource(fetcher, options)` is now recognized as the no-source form — the fetcher is not a tracked scope and may be async, so `noAsyncTrackedScope` no longer fires. The rule decides arg shapes by whether the second argument is a function.
